lib/schemes/oauth2.js

```diff
@@ -198,4 +202,85 @@ export default class Oauth2Scheme {
 
     return true // True means a redirect happened
   }
+
+  // ---------------------------------------------------------------
+  // Watch axios requests for token expiration
+  // Refresh tokens if token has expired
+  // ---------------------------------------------------------------
+
+  initializeTokenRefreshOnRequest () {
+    const { $axios } = this.$auth.ctx.app
+    let isRefreshing = false
+
+    $axios.onRequest(async config => {
+      let token = this.$auth.getToken(this.name)
+      let refreshToken = this.$auth.getRefreshToken(this.name)
+
+      // Token or "refresh token" does not exists
+      if (!token || !refreshToken || !token.length || !refreshToken.length) {
+        return config
+      }
+
+      // If already trying to refresh token, do not try again
+      if (isRefreshing) {
+        return config
+      }
+
+      // Time variables
+      let tokenExpiresAt = jwtDecode(token).exp * 1000
```

Comment:
This is implicitly considering that token is a (standard) JWT with exp field. Maybe better to mention it in docs?

Reply:
Also const :D

Reply:
Regarding using jwtDecode. I see in [the local refresh support PR](#361), the expires_in from the token response is used in combination with created_at to calculate the expiration time. We cannot use this here - as created_at is not part of the [oauth2 standard](https://tools.ietf.org/html/rfc6749#section-5.1).

Will document this :)